go-controller/pkg/util/kube.go

```diff
@@ -738,6 +738,16 @@ func PodTerminating(pod *corev1.Pod) bool {
 	return pod.DeletionTimestamp != nil
 }
 
+// PodNotReady checks if the pod is not ready
+func PodNotReady(pod *corev1.Pod) bool {
+	for _, condition := range pod.Status.Conditions {
+		if condition.Type == corev1.PodReady {
+			return condition.Status == corev1.ConditionFalse
+		}
+	}
+	return false
```

Comment:
also if no ready condition is found.. then we are treating it as "it is ready?" is that true to assume it?